### PR TITLE
Add useTabs hook

### LINKED ISSUE #1: Use `react-hooks-testing-library` for testing
We should change our local test helpers for the new `react-hooks-testing-library`.
I think it's cleverer than the used approach.

Repo: https://github.com/mpeyper/react-hooks-testing-library

What do we need to do?

- Add the `react-hooks-testing-library` package;
- Change the test boilerplate;
- Remove the test helper and its test file;

### LINKED ISSUE #3: useTabs ui hook
We need a hook to control the tabs usage logic.
It has to be simple and cover the main behaviors:
- Control current active tab
- Show and hide content events
- Dev shouldn't worry about styling

Example:

```javascript
const MyComponent = props => {
  const { activeTab, selectTab } = useTabs()
  return (
    <div className='tabs-container'>
      <div className='tabs'>
        {props.children.map((child, i) => (
          <a
            href='#!'
            className={`tab ${(activeTab === i && 'tab-active')}`}
            children={child.props.title}
            onClick={() => selectTab(i)}
          />
        ))}
      </div>
      <div className='tabs-content'>
        {props.children.filter((child, i) => activeTab === i)}
      </div>
    </div>
  )
}
```

### PR DESCRIPTION
**Pull request by jprask**
- Created useTabs hook and tests. Closes #3.
- Use `react-hooks-testing-library` for testing. Closes #1 

### Notes
Changed testing library to  react-hooks-testing-library in order to create tests for useTabs component.

### CODE REVIEW

File: tests/hooks/useTabs.test.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      const { disabledTabs } = result.current
+      expect(disabledTabs).toEqual([])
+    })
+
+    test('offering an isEnabled function', () => {
+      const config = {initialDisabledTabs: [1]}
+      const { result } = renderHook(() => useTabs(config))
+      const { isEnabled } = result.current
+      expect(isEnabled(0)).toBe(true)
+      expect(isEnabled(1)).toBe(false)
```
Comment (devthiago) [inline tests/hooks/useTabs.test.js:110]: Avoid more than one expect by test case.

```javascript
describe('isEnabled function response', () => {
  let current = null

  beforeAll(() => {
    const config = {initialDisabledTabs: [1]}
    const { result } = renderHook(() => useTabs(config))
    current = result.current
  })

  test('returns true if tab index is enabled', () => {
    const { isEnabled } = current
    expect(isEnabled(0)).toBe(true)
  })

  test('returns false if tab index is not enabled', () => {
    const { isEnabled } = current
    expect(isEnabled(1)).toBe(false)
  })
})
```

File: tests/hooks/useTabs.test.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    })
+  })
+
+})
```
Comment (devthiago) [inline tests/hooks/useTabs.test.js:113]: Be clear about which feature you are testing and the expected result:
```javascript
describe('useTabs()', () => {
  describe('activeTab integer response', () => {
    test('returns 0 if not receive an initial value', () => {
      const { result } = renderHook(() => useTabs())
      const { activeTab } = result.current
      expect(activeTab).toBe(0)
    })
  })

  describe('when selectTab function response is called', () => {
    test('activeTab is equal to the selectTab passed parameter', () => {
      const { result } = renderHook(() => useTabs())
      const { selectTab } = result.current
      const value = 1
      act(() => selectTab(value))
      const { activeTab } = result.current
      expect(activeTab).toBe(value)
    })
  })
})
```

**Review by devthiago — CHANGES_REQUESTED** [review]

File: docs/useTabs.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+id: use-tabs
+title: useTabs
+sidebar_label: useTabs
+---
+
+## Lorem
```
Comment (abnersajr) [inline docs/useTabs.md:7]: Does this text block in the right place? I guess this was not supposed to be here.

File: tests/hooks/useTabs.test.js (hunk 1 above)
Comment (jprask) [reply tests/hooks/useTabs.test.js:110]: I used the code you provided for those tests, thanks.

File: tests/hooks/useTabs.test.js (hunk 2 above)
Comment (jprask) [reply tests/hooks/useTabs.test.js:113]: Refactored the descriptions for more readability.

File: docs/useTabs.md (hunk 3 above)
Comment (jprask) [reply docs/useTabs.md:7]: That is placeholder for a more detailed description of the hook, perhaps with some examples, that we should write before releasing the library.

Comment (devthiago) [reply docs/useTabs.md:7]: I think we can do better when create the doc website. For while, it's fine.

**Review by devthiago — APPROVED** [review]
LGTM 👍

**Pull request merged**